app/src/main/java/com/nutomic/syncthingandroid/activities/FolderActivity.java

```diff
@@ -190,6 +205,39 @@ public void onCreate(Bundle savedInstanceState) {
         }
     }
 
+    /**
+     * Invoked after user clicked on the {@link mPathView} label.
+     */
+    @SuppressLint("InlinedAPI")
+    private void onPathViewClick() {
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.LOLLIPOP) {
+            startActivityForResult(FolderPickerActivity.createIntent(this, mFolder.path, null),
+                FolderPickerActivity.DIRECTORY_REQUEST_CODE);
+            return;
+        }
+
+        /**
+         * Determine the app's private data folder on external storage if present.
+         * e.g. "/storage/abcd-efgh/Android/com.nutomic.syncthinandroid/files"
+         */
+        File initialPath = null;
+        ArrayList<File> externalFilesDir = new ArrayList<>();
+        externalFilesDir.addAll(Arrays.asList(getExternalFilesDirs(null)));
+        externalFilesDir.remove(getExternalFilesDir(null));
+
+        // Display storage access framework directory picker UI.
+        Intent intent = new Intent(Intent.ACTION_OPEN_DOCUMENT_TREE);
+        if (externalFilesDir.size() > 0) {
+            initialPath = externalFilesDir.get(0);
+            Toast.makeText(this, getString(R.string.folder_hint_writeable_initial_folder, initialPath.getAbsolutePath()),
+                Toast.LENGTH_LONG).show();
+            intent.putExtra("android.provider.extra.INITIAL_URI", Uri.fromFile(initialPath).toString());
```

Comment:
This does not work on my Android version. SAF UI starts in the place where the user was the last time the UI was opened. Maybe a future android feature? I only made sure it doesn't crash.

Reply:
I still need to improve this. Testing on Android 8+ I discovered we need to put a parcelable here.